src/Features/LanguageServer/Protocol/Handler/References/FindAllReferencesHandler.cs

```
@@ -0,0 +1,172 @@
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Text.Adornments;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    // The VS LSP client supports streaming using IProgress<T> on various requests.
+    // However, this is not yet supported through Live Share, so deserialization fails on the IProgress<T> property.
+    // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1043376 tracks Live Share support for this (committed for 16.6).
+    [ExportLspMethod(LSP.Methods.TextDocumentReferencesName), Shared]
+    internal class FindAllReferencesHandler : IRequestHandler<LSP.ReferenceParams, object[]>
+    {
+        private readonly IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public FindAllReferencesHandler(IThreadingContext threadingContext)
+        {
+            _threadingContext = threadingContext;
+        }
+
+        public async Task<object[]> HandleRequestAsync(Solution solution, ReferenceParams referenceParams, ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
+        {
+            // The VS LSP client supports streaming using IProgress<T> on various requests.
+            // However, this is not yet supported through Live Share, so deserialization fails on the IProgress<T> property.
+            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1043376 tracks Live Share support for this (committed for 16.6).
+            var locations = ArrayBuilder<LSP.Location>.GetInstance();
+            var document = solution.GetDocumentFromURI(referenceParams.TextDocument.Uri);
+            if (document == null)
+            {
+                return locations.ToArrayAndFree();
```

these sorts of things worry me.  it seems like it indicates a bug, but we silently fail instead.  i'm not sure how  auser would tell what's going on.

---

Hmm... do you think calling .GetRequiredDocument instead would be better? But since it would throw if the document can't be found, I'm not sure that would be an improvement...